centreon/src/Core/Host/Infrastructure/Repository/DbWriteHostLogRepository.php

```diff
@@ -0,0 +1,275 @@
+                        throw new RepositoryException('Action log ID cannot be 0');
+                    }
+                    $actionLogChange->setId($actionLogChangeId);
+                    $this->writeActionLogRepository->addActionDetails($actionLogChange, $updatedHostDetails);
+                }
+            } else {
+                $actionLogChange = new ActionLog(
+                    self::HOST_OBJECT_TYPE,
+                    $host->getId(),
+                    $host->getName(),
+                    ActionLog::ACTION_TYPE_CHANGE,
+                    $this->contact->getId()
+                );
+                $actionLogChangeId = $this->writeActionLogRepository->addAction($actionLogChange);
+                if ($actionLogChangeId === 0) {
+                    throw new RepositoryException('Action log ID cannot be 0');
+                }
+                $actionLogChange->setId($actionLogChangeId);
+                $this->writeActionLogRepository->addActionDetails($actionLogChange, $updatedHostDetails);
+            }
+
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+
+            throw $ex;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function addParent(int $childId, int $parentId, int $order): void
+    {
+        $this->writeHostRepository->addParent($childId, $parentId, $order);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function deleteParents(int $childId): void
+    {
+        $this->writeHostRepository->deleteParents($childId);
+    }
+
+    /**
+     * @param NewHost $host
+     *
+     * @return array<string,int|bool|string>
+     */
+    private function getHostPropertiesAsArray(NewHost $host): array {
+        $hostPropertiesArray = [];
+        $hostReflection = new \ReflectionClass($host);
+
+        foreach ($hostReflection->getProperties() as $property) {
+            $value = $property->getValue($host);
+            if ($value === null) {
+                $value = '';
+            }
+
+            if ($value instanceof GeoCoords) {
+                $value = $value->__toString();
+            }
+
+            if ($value instanceof YesNoDefault) {
+                $value = YesNoDefaultConverter::toString($value);
+            }
+
+            if ($value instanceof SnmpVersion) {
+                $value = $value->value;
+            }
+
+            if (is_array($value)) {
+                if ($value === []) {
+                    $value = '';
+                } else {
+                    if (is_string($value[0])) {
+                        $value = '!' . implode(
+                            '!',
+                            str_replace(["\n", "\t", "\r"], ['#BR#', '#T#', '#R#'], $value)
+                        );
+                    }
+
+                    if ($value[0] instanceof HostEvent) {
+                        $value = HostEventConverter::toString($value);
+                    }
```

Comment:
enchainé avce des elseif plutôt non ? ca évite des sous if type Hadoken (si t'as la réf) 😃

Reply:
True, I will try to refacto in my next PRs